src/Disks/DiskLocal.cpp

```diff
@@ -188,6 +191,18 @@ void DiskLocal::moveDirectory(const String & from_path, const String & to_path)
     Poco::File(disk_path + from_path).renameTo(disk_path + to_path);
 }
 
+void DiskLocal::sync(const String & path) const
+{
+    String full_path = disk_path + path;
+    int fd = ::open(full_path.c_str(), O_RDONLY);
```

Comment:
It doesn't work this way, technically speaking.
https://stackoverflow.com/questions/37288453/calling-fsync2-after-close2

Reply:
It may work but it's not guaranteed...

Reply:
Yes, I've read this post and I am not sure about this code. But it's not clear how to fsync directories.
Looks like we should do something like:
```
mkdir("path", ...);
int dir_fd = open("path", O_RDONLY);

... write files ...
... fsync files...

fsync(dir_fd);
close(dir_fd);
```

Reply:
Yes, it's also not clear for me how to fsync directories. I think that your method is Ok for directories.

PS. Maybe better to separate fsync of directories to separate setting, because it's not required with typical filesystem options.

Reply:
Yes, it's valid, except the fact that fsyncing DIRFD can go in paralle with file data fsync.

DIRFD fsync is required if and only if THE DIRECTORY was changed - i.e. some files in the dir appeared or deleted or renamed.

Reply:
And also please change fsync to fdatasync everywhere.

Reply:
It's better not to reopen files for fsync purpose. But techincally it will work.